### ISSUE TITLE
Can't update/assign model with json value from params

### ISSUE THREAD

**Issue by samnang** [madeintandem/jsonb_accessor#80]
Here is my model:

```ruby
class Listing < ApplicationRecord
    jsonb_accessor :weekend_price, weekend_price_enabled: [:boolean, store_key: :enabled, default: false],
                                   weekend_price_value: [:integer, store_key: :value]

end
```

When I tried to update model from params:

```ruby

listing.update(weekend_price: { enabled: true })
```

Then I got an error: 

```
ActiveModel::MissingAttributeError: can't write unknown attribute `enabled`
```

I'm using `1.0.0.beta.6` version.

**Comment by crismali** [comment]
Looking at your code, it looks like what you need is:

```ruby
listing.update(weekend_price: { weekend_price_enabled: true })
```

**Comment by samnang** [comment]
Yep, that's what I tried and it works as well, but I feel it makes more sense to either do:

```ruby
listing.update(weekend_price: { enabled: true })

# or

listing.update(weekend_price_enabled: true)
```
I have tried to tweak the attribute setter's implementation to support API above:

```ruby
# jsonb_accessor/macro.rb
store_key_attributes = value.stringify_keys.slice(*empty_store_key_attributes.keys)
write_attribute(jsonb_attribute, empty_store_key_attributes.merge(store_key_attributes))

key_attributes = ::JsonbAccessor::QueryHelper.convert_store_keys_to_keys(value, names_to_store_keys)
empty_named_attributes.merge(key_attributes).each { |name, attribute_value| write_attribute(name, attribute_value) }

# jsonb_accessor/query_helper.rb
def convert_store_keys_to_keys(attributes, store_key_mapping)
  attributes.each_with_object({}) do |(name, value), new_attributes|
    key = store_key_mapping.key(name.to_s)
    new_attributes[key] = value
  end
end
```

Let me know if you think the change is good. If so, I will submit PR and add/update some tests to cover this functionality.

**Comment by crismali** [comment]
Does `listing.update(weekend_price_enabled: true)` not work?

**Comment by crismali** [comment]
Sorry it took me a bit to get back to you. I think this is a sensible enhancement. Definitely open to a PR :)

**Comment by rainerborene** [comment]
Another issue related to this is that I can't reset the `jsonb` column to `{}` without having all fields set to null.

**Comment by haffla** [comment]
I think this has been fixed a long time ago. Feel free to reopen.

**Comment by joelvh** [comment]
@haffla this is the issue that brought me back to this repo the other week - it's an issue, at least in the version of the gem that was published until recently. Was a fix for this included in the gem you just published?

**Comment by haffla** [comment]
The problem is that I don't understand what's the main issue here. You can certainly update the jsonb property with a hash object directly and as of today the issue mentioned by @rainerborene above is fixed.

So is the problem that the update doesn't work if the `store_key` property is used and the payload of the update has the `store_key`? I haven't actually looked into that, sorry. Happy to accept a PR / will try to fix it soon.

**Comment by joelvh** [comment]
@haffla yeah, i think you've got it. What we ran into is

```ruby
class Order < ApplicationRecord
  jsonb_accessor :attributions,
                 attribution_source: [:string, { store_key: :source }],
                 attribution_channel: [:string, { store_key: :channel }]
end

o = Order.new
o.attributions = { source: 'a', channel: 'b' }
# => ActiveModel::MissingAttributeError: can't write unknown attribute `source`
# => from /Users/joel/.rbenv/versions/2.6.7/lib/ruby/gems/2.6.0/gems/activemodel-6.1.3.2/lib/active_model/attribute.rb:207:in `with_value_from_database'
```

In this case, it would seem intuitive to use the stored keys when setting the whole JSON object. For us, this was a case where we were just trying to copy the column data from one instance to another and got that error.

I'll see if I can devise a PR for this.

**Comment by haffla** [comment]
@joelvh I think I've got something already. Gonna push now.

**Comment by haffla** [comment]
I've released version [1.3.2](https://rubygems.org/gems/jsonb_accessor/versions/1.3.2) which fixes the issue.
